samples/add-feature-collection-layer-from-table/src/main/java/com/esri/arcgismaps/sample/addfeaturecollectionlayerfromtable/components/AddFeatureCollectionLayerFromTableViewModel.kt

```diff
@@ -0,0 +1,237 @@
+import com.arcgismaps.data.FeatureCollectionTable
+import com.arcgismaps.data.Field
+import com.arcgismaps.data.FieldType
+import com.arcgismaps.geometry.Polygon
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.Polyline
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.FeatureCollectionLayer
+import com.arcgismaps.mapping.symbology.SimpleFillSymbol
+import com.arcgismaps.mapping.symbology.SimpleFillSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbol
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleRenderer
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.launch
+
+/**
+ * ViewModel which creates three in-memory FeatureCollectionTables (point, polyline, polygon),
+ * adds features to them, creates a FeatureCollection and displays it on the map as a
+ * FeatureCollectionLayer.
+ */
+class AddFeatureCollectionLayerFromTableViewModel(app: Application) : AndroidViewModel(app) {
+
+    // ArcGIS map exposed as mutable state for Compose.
+    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISOceans).apply {
+            initialViewpoint = Viewpoint(latitude = 8.849289, longitude = -79.497238, scale = 1e6)
+     }
+
+    // Message dialog view model for error handling
+    val messageDialogVM = MessageDialogViewModel()
+
+    init {
+        // Load the map and then build the feature collection layer
+        viewModelScope.launch {
+            arcGISMap.load()
+                .onFailure { messageDialogVM.showMessageDialog(it) }
+                .onSuccess {
+                // Create feature collection tables and add them to the map
+                createAndAddFeatureCollectionLayer()
+            }
+        }
+    }
+
+    private fun createAndAddFeatureCollectionLayer() {
+        try {
+            // Create each feature collection table
+            val pointsTable = createPointsCollectionTable()
+            val linesTable = createLinesCollectionTable()
+            val polygonsTable = createPolygonsCollectionTable()
+
+            // Create a FeatureCollection from the tables
+            val featureCollection = FeatureCollection(featureCollectionTables = listOf(pointsTable, linesTable, polygonsTable))
+
+            // Create a layer from the feature collection and add it to the map
+            val featureCollectionLayer = FeatureCollectionLayer(featureCollection = featureCollection)
+            arcGISMap.operationalLayers.add(featureCollectionLayer)
+        } catch (ex: Exception) {
+            // Surface any unexpected exceptions in a dialog
+            messageDialogVM.showMessageDialog(ex)
+        }
```

Comment:
Any reason why the try/catch is needed here? I would opt to remove it if an error is not expected. If yes, I would opt to use `onFailure` or `getOrElse` instead.

Reply:
removed the try catch